wfe2/wfe.go

```diff
@@ -350,12 +360,17 @@ func (wfe *WebFrontEndImpl) Handler(stats prometheus.Registerer) http.Handler {
 	wfe.HandleFunc(m, directoryPath, wfe.Directory, "GET", "POST")
 	wfe.HandleFunc(m, newNoncePath, wfe.Nonce, "GET", "POST")
 	// POST-as-GETable ACME endpoints
-	// TODO(@cpu): After November 1st, 2019 support for "GET" to the following
+	// TODO(@cpu): After November 1st, 2020 support for "GET" to the following
 	// endpoints will be removed, leaving only POST-as-GET support.
 	wfe.HandleFunc(m, orderPath, wfe.GetOrder, "GET", "POST")
 	wfe.HandleFunc(m, authzv2Path, wfe.AuthorizationV2, "GET", "POST")
 	wfe.HandleFunc(m, challengev2Path, wfe.ChallengeV2, "GET", "POST")
 	wfe.HandleFunc(m, certPath, wfe.Certificate, "GET", "POST")
+	// Boulder-specific ACME GET-able endpoints
+	wfe.HandleFunc(m, getOrderPath, wfe.GetOrder, "GET")
+	wfe.HandleFunc(m, getAuthzv2Path, wfe.AuthorizationV2, "GET")
+	wfe.HandleFunc(m, getChallengev2Path, wfe.ChallengeV2, "GET")
```

Comment:
The `wfe.AuthorizationV2` and `wfe.ChallengeV2` functions are being registered for GET in this branch but I don't think they have the same diffs you're adding for `wfe.GetOrder` and `wfe.Certificate` to ensure only resources older than the `staleTimeout` can be fetched with the handlers. Did I miss something?

I think it would be nice to have test coverage for the Boulder-specific GET-able authz and challenge endpoints too. Everything else looks good to me.

Reply:
@cpu I don't think we need to re-test the same functions on other endpoints. But if you really want tests, I could duplicate them.

Reply:
Added challenge test, where's the test for authz?

Reply:
Looks like it's tested directly, without routing: 
![image](https://user-images.githubusercontent.com/19504461/71303701-6fd94800-23cd-11ea-847a-832380463742.png)

Reply:
This feedback was marked resolved but the `wfe.Certificate` and `wfe.GetOrder` endpoints still do not enforce that only stale resources be accessed when using the non-standard GET method.